src/components/views/voip/CallView.tsx

```diff
@@ -14,412 +14,57 @@ See the License for the specific language governing permissions and
-                    connectDeferred.resolve();
-                } catch (e) {
-                    connectDeferred.reject(e);
-                }
-            }
-        })();
-    }, [call, connectDeferred]);
-
-    return (
-        <div className="mx_CallView" role={role}>
-            {connected ? null : <Lobby room={room} connect={connect} />}
-            {call !== null && (
-                <AppTile
-                    app={call.widget}
-                    room={room}
-                    userId={cli.credentials.userId!}
-                    creatorUserId={call.widget.creatorUserId}
-                    waitForIframeLoad={call.widget.waitForIframeLoad}
-                    showMenubar={false}
-                    pointerEvents={resizing ? "none" : undefined}
-                />
-            )}
-        </div>
-    );
-};
+import { SdkContextClass } from "../../../contexts/SDKContext";
 
 interface JoinCallViewProps {
     room: Room;
     resizing: boolean;
     call: Call;
+    skipLobby?: boolean;
     role?: AriaRole;
 }
 
-const JoinCallView: FC<JoinCallViewProps> = ({ room, resizing, call, role }) => {
+const JoinCallView: FC<JoinCallViewProps> = ({ room, resizing, call, skipLobby, role }) => {
     const cli = useContext(MatrixClientContext);
-    const connected = isConnected(useConnectionState(call));
-    const members = useParticipatingMembers(call);
-    const joinCallButtonDisabledTooltip = useJoinCallButtonDisabledTooltip(call);
-
-    const connect = useCallback(async (): Promise<void> => {
-        // Disconnect from any other active calls first, since we don't yet support holding
-        await Promise.all([...CallStore.instance.activeCalls].map((call) => call.disconnect()));
-        await call.connect();
-    }, [call]);
 
-    // We'll take this opportunity to tidy up our room state
     useEffect(() => {
+        // We'll take this opportunity to tidy up our room state
         call.clean();
     }, [call]);
 
-    let lobby: JSX.Element | null = null;
-    if (!connected) {
-        let facePile: JSX.Element | null = null;
-        if (members.length) {
-            const shownMembers = members.slice(0, MAX_FACES);
-            const overflow = members.length > shownMembers.length;
+    useEffect(() => {
+        // Always update the widget data so that we don't ignore "skipLobby" accidentally.
+        (call.widget.data ?? { skipLobby }).skipLobby = skipLobby;
+    }, [call.widget.data, skipLobby]);
 
-            facePile = (
-                <div className="mx_CallView_participants">
-                    {_t("voip|n_people_joined", { count: members.length })}
-                    <FacePile members={shownMembers} size="24px" overflow={overflow} />
-                </div>
-            );
+    useEffect(() => {
+        if (call.connectionState === ConnectionState.Disconnected) {
+            // immediately connect (this will start the lobby view in the widget)
+            call.connect();
```

Comment:
Connect is actually not the correct name for this anymore. Connect now has become sth like `start`.
I was considering renaming it or exploring starting the lobby from a different step in the Call lifecycle.
Since the widget is now responsible for the initiation and the execution of connecting having this function on the Call object is not a good fit anymore. But since Call is the abstraction for other calls as well I did not want to rename it.

Reply:
Rename to `start()` and updated related comments so it becomes more clear what is happening.